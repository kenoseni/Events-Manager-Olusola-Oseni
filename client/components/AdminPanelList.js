import React, {Component} from 'react';
import PropTypes from 'prop-types';
import AdminPanel from './AdminPanel';
import NavBar from './Navbar';

class AdminPanelList extends Component {
  render () {
    const { users, match, history, searchForCenters } = this.props;
    const { count, limit } = this.props.user;
    return (
      <div style={{paddingTop: '90px'}}>
        <NavBar home='Home' admin='Admin' centers='Centers' events='Events' page='Admin' logout='Log Out' match={match} searchForCenters={searchForCenters} />
        <div className="container">
          <div className="row" id="users">
            <AdminPanel users={users} match={match} history={history} count={count} limit={limit} {...this.props}/>  
          </div>
        </div>
      </div>
    )
  }
}
AdminPanelList.propTypes = {
  users: PropTypes.array,
  match: PropTypes.object.isRequired
}
export default AdminPanelList
