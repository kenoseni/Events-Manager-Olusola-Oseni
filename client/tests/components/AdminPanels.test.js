import React from 'react';
import { configure, shallow } from 'enzyme';
import sinon from 'sinon';
import Adapter from 'enzyme-adapter-react-16';
import Connected, { AdminPanels } from '../../components/AdminPanels';
import AdminPanelList from '../../components/AdminPanelList';

configure({ adapter: new Adapter() });

const mockgetAllUsers = jest.fn();

describe('<AdminPanels />', () => {
  let props = {
    user: {
      users: {}
    },
    isAuthenticated: true,
    isAdmin: true,
    userDetails: {
      count: 0
    },
    history: {
      location: {
        search: '/page=1'
      }
    }
  }
  const wrapper = shallow(<AdminPanels
    count={props.userDetails.count}
    history={props.history}
    getAllUsers={mockgetAllUsers}
    {...props} />
  )
  it('should render a <AdminPanelList /> component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('it should call getAllUsers function', () => {
    wrapper.setProps({
      ...wrapper.props,
      isAuthenticated: false,
      isAdmin: false
    })
    expect(wrapper.find('.alert').length).toBe(1)
  });
});
