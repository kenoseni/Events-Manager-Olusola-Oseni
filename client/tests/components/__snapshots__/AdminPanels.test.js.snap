// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminPanels /> should render a <AdminPanelList /> component 1`] = `
<div>
  <AdminPanelList
    count={0}
    getAllUsers={
      [MockFunction] {
        "calls": Array [
          Array [
            "1",
          ],
        ],
      }
    }
    history={
      Object {
        "location": Object {
          "search": "/page=1",
        },
      }
    }
    isAdmin={true}
    isAuthenticated={true}
    user={
      Object {
        "users": Object {},
      }
    }
    userDetails={
      Object {
        "count": 0,
      }
    }
    users={Object {}}
  />
</div>
`;
